dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageLoader.java

```diff
@@ -0,0 +1,219 @@
+            .find(htmlPage);
+        try {
+            return buildStream(htmlPage, identifier, mode, filePath);
+        } catch (Exception e) {
+            Logger.error(this.getClass(), e.getMessage(), e);
+            throw new DotRuntimeException(e.getMessage());
+        }
+    }
+
+
+    public InputStream buildStream(IHTMLPage htmlPage, Identifier identifier, PageMode mode, final String filePath)
+            throws DotDataException, DotSecurityException {
+        String folderPath = mode.name() + File.separator;
+        InputStream result;
+        StringBuilder sb = new StringBuilder();
+
+        ContentletAPI conAPI = APILocator.getContentletAPI();
+        Template cmsTemplate = APILocator.getHTMLPageAssetAPI()
+            .getTemplate(htmlPage, mode.showLive);
+
+
+        User sys = APILocator.systemUser();
+
+
+        if (cmsTemplate == null || !InodeUtils.isSet(cmsTemplate.getInode())) {
+            Logger.error(This.class, "PAGE DOES NOT HAVE A VALID TEMPLATE (template unpublished?) : page id "
+                    + htmlPage.getIdentifier() + ":" + identifier.getURI());
+        }
+
+
+
+        // List of tags found in this page
+        List<Tag> pageFoundTags = new ArrayList<>();
+
+        // Check if we want to accrue the tags of this HTMLPage contentlet
+        if (Config.getBooleanProperty("ACCRUE_TAGS_IN_PAGES", true)) {
+
+            List<Tag> htmlPageFoundTags = APILocator.getTagAPI()
+                .getTagsByInode(htmlPage.getInode());
+            if (htmlPageFoundTags != null && !htmlPageFoundTags.isEmpty()) {
+                pageFoundTags.addAll(htmlPageFoundTags);
+            }
+        }
+
+
+
+        sb.append("#set($dotPageContent = $dotcontent.find(\"" + htmlPage.getInode() + "\" ))");
+
+
+        // set the host variables
+
+        Host host = APILocator.getHTMLPageAssetAPI()
+            .getParentHost(htmlPage);
+        sb.append("#if(!$doNotParseTemplate)")
+            .append("#parse('")
+            .append(folderPath)
+            .append(host.getIdentifier())
+            .append(".")
+            .append(VelocityType.SITE.fileExtension)
+            .append("')")
+            .append("#end");
+
+
+        sb.append(new PageContextBuilder(htmlPage, sys, mode).asString());
+
+
+        // Now we need to use the found tags in order to accrue them each time this page is visited
+        if (!pageFoundTags.isEmpty()) {
+            // Velocity call to accrue tags on each request to this page
+            sb.append("$tags.accrueTags(\"" + TagUtil.tagListToString(pageFoundTags) + "\" )");
+        }
+        if (htmlPage.isHttpsRequired()) {
+            sb.append(" #if(!$ADMIN_MODE  && !$request.isSecure())");
+            sb.append("#if($request.getQueryString())");
+            sb.append("#set ($REDIRECT_URL = \"https://${request.getServerName()}$request.getAttribute('javax.servlet.forward.request_uri')?$request.getQueryString()\")");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 171 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)